pkg/operation/common/utils.go

```diff
@@ -197,6 +197,49 @@ func DeleteAlertmanager(ctx context.Context, k8sClient client.Client, namespace
 	return kutil.DeleteObjects(ctx, k8sClient, objs...)
 }
 
+// DeleteGrafana deletes the monitoring stack for the shoot owner.
+func DeleteGrafana(ctx context.Context, k8sClient kubernetes.Interface, namespace string) error {
+	if k8sClient == nil {
+		return fmt.Errorf("require kubernetes client")
+	}
+
+	deleteOptions := []client.DeleteAllOfOption{
+		client.InNamespace(namespace),
+		client.MatchingLabels{
+			"component": "grafana",
+		},
+	}
+
+	if err := k8sClient.Client().DeleteAllOf(ctx, &appsv1.Deployment{}, append(deleteOptions, client.PropagationPolicy(metav1.DeletePropagationForeground))...); client.IgnoreNotFound(err) != nil {
```

Comment:
I don't think a `DeleteAllOf` (delete collection) call can return a not found error. Ref https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#delete-collection-pod-v1-core.

Do you see any other usage of `DeleteAllOf` gardener/gardener to ignore not found errors?

```suggestion
	if err := k8sClient.Client().DeleteAllOf(ctx, &appsv1.Deployment{}, append(deleteOptions, client.PropagationPolicy(metav1.DeletePropagationForeground))...); err != nil {
```

Reply:
@wyb1 and I tested it and actually a `not found` error is returned which makes the consecutive reconciliations fail.
I'm not sure if this document lists the possible errors that can be returned by the function: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#delete-pod-v1-core

```
2022-10-21T08:42:30.013Z ERROR controller.shoot.shoot Error reconciling request
{name: local2,
namespace: garden-local,
error: Flow "Shoot cluster reconciliation" encountered task errors:
[task "Reconciling Grafana for Shoot in Seed for the logging stack" failed:
retry failed with context deadline exceeded,
last error: services "grafana-operators" not found task "Deploying Shoot monitoring stack in Seed" failed:
retry failed with context deadline exceeded,
last error: secret "observability-ingress-users" not found]}
```

Reply:
You should be able to remove the `ignore not found` from `DeleteAllOf` calls. You problem was rather that in https://github.com/gardener/gardener/pull/6749/commits/882885cc4ff9eb9914f1ce2d0f68a9c28968eb2c you removed the `client.IgnoreNotFound` from L236 which is actually not a `DeleteAllOf` call but a `Delete` call. You should preserve the  `client.IgnoreNotFound` there.

Reply:
I will also try to test on my end in a while

Reply:
Thank you for finding out the issue. I repushed the commit without the unneeded change: https://github.com/gardener/gardener/pull/6749/commits/770eaa160eee54374b4cc8761142355b8c3e3446

Reply:
Thanks for incorporating the suggestion! The latest revision works on my end as well! 👍